index.js

```diff
@@ -4,14 +4,16 @@ function hook(type, opts, cb) {
 	if (typeof opts !== 'object') {
 		cb = opts;
 		opts = {};
+		opts.silent = true;
+	} else if (opts && typeof opts.silent !== 'boolean') {
+		opts.silent = true;
```

Comment:
You should never modify user-supplied objects. This can all kinds of issues when the object is reused between calls, something users could do.